File: two-sum.php
<?php

function twoSum($target, $nums) {
    $pair = [];
    
    foreach ($nums as $k => $v) {
        
        $vv = $target - $v;
        
        $kk = array_search($vv, $pair);
        
        if ($kk) {
            return [$kk, $k];
        }
        $pair[$k] = $v;
    }
}

$target = 6;
$nums = [2, 1, 5, 8];

var_dump(twoSum($target, $nums));

File: two-sum/two-sum.py
target = 3
nums = [2, 1, 5, 8]
pair = {}

# WIP


def two_sum():
    for i, num in enumerate(nums):
        vv = target - num
        if vv in pair:
            return [pair[vv], i]

        pair[i] = num
